src/misc/breaking-bottles.ts

```diff
@@ -0,0 +1,109 @@
+import { Material, Particle, Sound } from 'org.bukkit';
+import { Damageable, Item, Snowball } from 'org.bukkit.entity';
+import { ProjectileHitEvent } from 'org.bukkit.event.entity';
+import { PlayerDropItemEvent } from 'org.bukkit.event.player';
+import { ItemStack } from 'org.bukkit.inventory';
+import { PotionMeta } from 'org.bukkit.inventory.meta';
+import { Location } from 'org.bukkit';
+import { Vector } from 'org.bukkit.util';
+
+// Materials for different colored potion particles
+const particles: Record<string, Material> = {
+  FIRE_RESISTANCE: Material.YELLOW_WOOL,
+  SLOW_FALLING: Material.WHITE_WOOL,
+  STRENGTH: Material.RED_WOOL,
+};
+
+const BREAK_VELOCITY = 0.12;
+const HIT_DAMAGE = 0.2;
+
+/**
+ *  Plays bottle break sound and particles at location of dropped bottle
+ * @param drop Dropped bottle
+ */
+function playBottleBreakEffects(loc: Location, item: ItemStack) {
+  const meta = item.itemMeta as PotionMeta;
+  const type = meta.basePotionData ? meta.basePotionData.type : null;
+
+  const itemParticle = (type && particles[`${type}`]) || null;
+  const waterParticle = item.type === Material.POTION;
+
+  loc.world.playSound(loc, Sound.BLOCK_GLASS_BREAK, 1, 1);
+
+  // Default broken glass particle
+  loc.world.spawnParticle(Particle.ITEM_CRACK, loc, 8, 0.3, 0.5, 0.3, 0, item);
+
+  // Special potion particle (If potion has been assigned extra material for color)
+  if (itemParticle) {
+    loc.world.spawnParticle(
+      Particle.ITEM_CRACK,
+      loc,
+      8,
+      0.2,
+      0.5,
+      0.2,
+      0,
+      new ItemStack(itemParticle),
+    );
+  }
+
+  // Water splash particle (If bottle isn't empty)
+  if (waterParticle) {
+    loc.world.spawnParticle(Particle.WATER_SPLASH, loc, 10, 0.2, 0.5, 0.2, 0);
+  }
+}
+
+/**
+ * Checks if dropped bottle has hit the ground and breaks the
+ * bottle if hit velocity is high enough
+ * @param drop Dropped bottle
+ */
+async function breakAfterFall(drop: Item) {
+  let previousY = drop.velocity.y;
+  let currentVel = 0;
+  await wait(100, 'millis'); // Wait for bottle to move before checks
+
+  while (drop && drop.velocity.y < previousY) {
```

Comment:
Should we make some kind of limit on how long this loop will be going on? I think items can be falling a very long time, if they are dropped in the "bubble water". So maybe check if the block is water or set max falling time (maybe 20 seconds)?